spec/factories/bulk_update/trainee_uploads.rb

```diff
@@ -10,8 +10,11 @@
 
 FactoryBot.define do
   factory :bulk_update_trainee_upload, class: "BulkUpdate::TraineeUpload" do
+    transient do
+      user { association(:user) }
+    end
+
```

Comment:
```suggestion
    user
```

I think you can just do this

Reply:
The reason I used a transient attribute is because there isn't actually an association with the user. I use it further down in the `after(:build)` callbacks to assign a `submitted_by` user in some of the states.

Reply:
oh right,

```
submitted_by { association(:user) }
```

instead of 

```
      after(:build) do |bulk_update_trainee_upload, evaluator|
        bulk_update_trainee_upload.submitted_by ||= evaluator.user
      end
```
then we can simply remove 
```suggestion

```

Reply:
Refactored https://github.com/DFE-Digital/register-trainee-teachers/pull/4852/commits/20a926124614056d50800139a4340e5aedd3a96c